app/scripts/utils.js

```diff
@@ -0,0 +1,52 @@
+// calculate hex cell width as css calc() function
+export const getCellWidth = (total) => `calc(${1 / (total) * 100}% + 2px)`;
+
+export const getSpannedTitle = (title) => {
+  return title.split('.').map(span => `<span>${span}</span>`).join('.');
+}
+
+export const getMedia = (media) => {
+  return media.reduce((prev, curr) => 
+  window.innerWidth >= curr.breakpoint ? curr : prev, media[0]);
+}
+
+export const getSvgIcon = (type, iconId, className, width, height) => {
+  return `
+    <svg
+      class=${className}
+      viewBox="0 0 ${width} ${height}"
+      ${iconId ? ` data-id=${iconId}` : ''}
+    >
+      ${type === 'blank' && '<path d="M100 28.867v57.735L50 115.47 0 86.602V28.867L50 0z"/>'}
+      ${type === 'label' && `<use href="svg/vectors.svg#${iconId}"></use>`}
+      ${type === 'logo' && `
+        <polyline class="logo__triangle" points="88.971,27.5 50,95 11.029,27.5 88.971,27.5" />
+        <polyline class="logo__outer" points="88.971,27.5 88.971,72.5 50,95" />
+        <polyline class="logo__outer" points="50,95 11.029,72.5 11.029,27.5 " />
+        <polyline class="logo__outer" points="11.029,27.5 50,5 88.971,27.5" />
+        <polyline class="logo__inner" points="88.971,27.5 50,50" />
+        <polyline class="logo__inner" points="50,95 50,50" />
+        <polyline class="logo__inner" points="11.029,27.5 50,50" />
+        <circle class="logo__circle" cx="88.971" cy="27.5" r="5" />
+        <circle class="logo__circle" cx="11.029" cy="27.5" r="5" />
+        <circle class="logo__circle" cx="50" cy="95" r="5" />
+        <circle class="logo__circle" cx="50" cy="50" r="5" />
+      `}      
+    </svg>
+  `
+}
+
+export const getTotalColumns = (data) => {
+  return data.reduce((max, curr) => curr.length > max.length ? curr : max).length;
+}
+
+export const createElement = (className = '', type = 'div', content = '') => {
+  const element = document.createElement(type);
+  element.className = className;
+  element.innerHTML = content;
+  if (type === 'a') {
+    element.target = '_blank';
+    element.rel = 'noopener noreferrer';
```

Comment:
Why do we want all links to be `noreferrer`? What if we use `createElement` function to the link redirecting appsilon.com? Then we won't be able to see in Google Analytics referrals from shiny.tools. I think we should remove this parameter, but let me know if there is any specific reason.

Reply:
Good point!
1. For links with target = '_blank' I used to add 'noopener noreferrer' due to security reasons (e.g. https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/), but the downside is no analytics available. Maybe only 'noopener' is safe enough?
2. This function assumes only outgoing link on the page, so that's why I used target = "_blank" for all links, but I will add possibility to have internal links without opening it in a new card.